File: app/src/main/java/com/fstyle/structure_android/screen/searchresult/SearchResultModule.java
package com.fstyle.structure_android.screen.searchresult;

import android.app.Activity;
import android.support.annotation.NonNull;
import com.fstyle.structure_android.data.model.User;
import com.fstyle.structure_android.utils.Constant;
import com.fstyle.structure_android.utils.dagger.ActivityScope;
import dagger.Module;
import dagger.Provides;
import java.util.ArrayList;

/**
 * Created by Sun on 3/19/2017.
 */

@Module
public class SearchResultModule {

    private Activity mActivity;

    public SearchResultModule(@NonNull Activity activity) {
        this.mActivity = activity;
    }

    @ActivityScope
    @Provides
    public SearchResultContract.ViewModel provideViewModel(SearchResultContract.Presenter presenter,
            SearchResultAdapter adapter) {
        return new SearchResultViewModel(presenter, adapter);
    }

    @ActivityScope
    @Provides
    public SearchResultContract.Presenter providePresenter() {
        return new SearchResultPresenter();
    }

    @ActivityScope
    @Provides
    public SearchResultAdapter provideSearchResultAdapter() {
        ArrayList<User> users =
                mActivity.getIntent().getParcelableArrayListExtra(Constant.ARGUMENT_LIST_USER);
        return new SearchResultAdapter(mActivity, users);
    }
}

File: app/src/main/java/com/fstyle/structure_android/screen/searchresult/SearchResultModule.kt
package com.fstyle.structure_android.screen.searchresult

import android.app.Activity
import com.fstyle.structure_android.data.model.User
import com.fstyle.structure_android.utils.Constant
import com.fstyle.structure_android.utils.dagger.ActivityScope
import com.fstyle.structure_android.utils.navigator.Navigator
import dagger.Module
import dagger.Provides

/**
 * Created by Sun on 3/19/2017.
 */

@Module
class SearchResultModule(private val mActivity: Activity) {

  @ActivityScope
  @Provides
  fun provideViewModel(presenter: SearchResultContract.Presenter,
      adapter: SearchResultAdapter, navigator: Navigator): SearchResultContract.ViewModel {
    return SearchResultViewModel(presenter, adapter,navigator)
  }

  @ActivityScope
  @Provides
  fun providePresenter(): SearchResultContract.Presenter {
    return SearchResultPresenter()
  }

  @ActivityScope
  @Provides
  fun provideSearchResultAdapter(): SearchResultAdapter {
    val users = mActivity.intent.getParcelableArrayListExtra<User>(
        Constant.ARGUMENT_LIST_USER)
    return SearchResultAdapter(mActivity, users)
  }

  @ActivityScope
  @Provides
  fun provideNavigator(): Navigator {
    return Navigator(mActivity)
  }
}
